speechbrain/lobes/models/kmeans.py

```diff
@@ -0,0 +1,168 @@
+"""K-means implementation.
+
+Authors
+* Luca Della Libera 2024
+"""
+
+import joblib
+import torch
+from sklearn.cluster import MiniBatchKMeans
```

Comment:
It would be cool to have this also in PyTorch, like this https://github.com/jokofa/torch_kmeans

But for practical reasons I guess it makes sense to keep it this way, though we will want to add it to `integrations` most likely.

Reply:
Public PyTorch implementations typically implement K-means instead of mini-batch K-means, this requires several modifications. I had a PR for that (https://github.com/speechbrain/speechbrain/pull/2411), but it was decided to move on with the scikit-learn implementation. It would be hard now to switch while maintaining backward compatibility, since we have a lot of models trained using the scikit-learn recipe.